tutorials/tensordictmodule.ipynb

```diff
@@ -0,0 +1,1020 @@
+      "File \u001b[0;32m~/anaconda3/envs/torch_rl/lib/python3.9/site-packages/torch/nn/modules/linear.py:114\u001b[0m, in \u001b[0;36mLinear.forward\u001b[0;34m(self, input)\u001b[0m\n\u001b[1;32m    113\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mforward\u001b[39m(\u001b[38;5;28mself\u001b[39m, \u001b[38;5;28minput\u001b[39m: Tensor) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m Tensor:\n\u001b[0;32m--> 114\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mF\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mlinear\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43minput\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mweight\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbias\u001b[49m\u001b[43m)\u001b[49m\n",
+      "\u001b[0;31mRuntimeError\u001b[0m: batched == nullptr INTERNAL ASSERT FAILED at \"/private/var/folders/fn/c72nxv0x0xj4bzdgq3b0r86r0000gn/T/pip-req-build-i1bssf2g/functorch/csrc/Interpreter.cpp\":95, please report a bug to PyTorch. "
+     ]
+    }
+   ],
+   "source": [
+    "num_models = 10\n",
+    "\n",
+    "tensordict = TensorDict({\"a\": torch.randn(5, 3)}, batch_size=[5])\n",
+    "\n",
+    "splitlinear_models = [\n",
+    "    TensorDictModule(\n",
+    "        nn.Linear(3, 10), in_keys=[\"a\"], out_keys=[\"output\"]\n",
+    "    )\n",
+    "    for _ in range(num_models)\n",
+    "]\n",
+    "\n",
+    "\n",
+    "def transpose_stack(tuple_of_tuple_of_tensors):\n",
+    "    tuple_of_tuple_of_tensors = tuple(zip(*tuple_of_tuple_of_tensors))\n",
+    "    results = tuple(\n",
+    "        torch.stack(shards).detach()\n",
+    "        for shards in tuple_of_tuple_of_tensors\n",
+    "    )\n",
+    "    return results\n",
+    "\n",
+    "\n",
+    "func = splitlinear_models[0].make_functional_with_buffers()[0]\n",
```

Comment:
can you comment a bit this code and say what you're doing?
Otherwise we leave it to the user to decipher it...